drivers/kw41zrf/kw41zrf_xcvr.c

```diff
@@ -0,0 +1,912 @@
+/*
+ * The Clear BSD License
+ * Copyright 2016-2017 NXP
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted (subject to the limitations in the
+ * disclaimer below) provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * NO EXPRESS OR IMPLIED LICENSES TO ANY PARTY'S PATENT RIGHTS ARE
+ * GRANTED BY THIS LICENSE. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT
+ * HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+ * WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+ * BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+ * OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+ * IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @ingroup     drivers_kw41zrf
+ * @{
+ * @file
+ * @brief       NXP KW41Z XCVR module initialization and calibration of kw41zrf driver
+ *
+ * @author      Joakim Nohlgård <joakim.nohlgard@eistec.se>
+ * @}
+ */
+#include <stdint.h>
+#include <string.h>
+#include <errno.h>
+
+#include "log.h"
+#include "bit.h"
+#include "kw41zrf.h"
+#include "vendor/XCVR/MKW41Z4/fsl_xcvr.h"
+#include "vendor/XCVR/MKW41Z4/ifr_radio.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+/* The implementations for these functions are taken from the vendor provided XCVR
+ * driver from NXP MCUXpresso. The code has been refactored to eliminate a lot
+ * of preprocessor conditionals. */
```

Comment:
I certainly agree in general. My understanding from reading the dialog in #7107 is that this was necessary or at least prohibitively inconvenient to avoid.

The version (which I can't confirm personally) is mentioned in the PR only. I'll add it to this comment.